### ISSUE TITLE
The libraries and costumes dialogs are so CPU-intensive that they crash some old computers.

### ISSUE THREAD

**Issue by kephas** [jmoenig/Snap--Build-Your-Own-Blocks#2139]
On several computers (either an old one with not much CPU to begin with or with a new one that's already under a certain load), both dialogs tend to be so CPU-intensive that the system starts to lag (sometimes with warnings from the browser that the page is responsible) and, in a few cases, froze entirely.

**Comment by bromagosa** [comment]
Yep. Someone needs to write a categorized costume browser. There was a PR addressing this but it was a bit messy because it also changed several things in the UI like font sizes and other unrelated stuff.

**Comment by kephas** [comment]
I doubt it is just a question of the number of costumes, because it also happens with libraries.

**Comment by brianharvey** [comment]
Weren't we going to solve this problem by keeping thumbnails?  For libraries, the time is going into preparing the pictures of blocks, I'm pretty sure.

**Comment by kephas** [comment]
Switching to "flat design" indeed made the library UI pretty fast, so i guess you're right, @brianharvey.

**Comment by cycomachead** [comment]
Did you already view the libraries in "regular" mode before you switched? Once you open the libraries dialog and view a library, viewing it again will be much faster.

Opening the libraries dialog alone should be quick, and it should load the list of libraries on the left quickly as well. However, the slow part is that when loading a library the whole thing has to be downloaded, and parsed and that's what's slow. But, once you import the library, it will be pretty fast, because Snap! already did the work to load the library.
